### PR TITLE
fix:blocksyncer oom

### PR DESCRIPTION
**Pull request by constwz**
### Description

add a description of your changes here...

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

File: modular/blocksyncer/blocksyncer_options.go [hunk 1]
```diff
@@ -134,29 +122,19 @@ func (b *BlockSyncerModular) initClient() error {
...
+				err = module.AutoMigrate()
+			} else {
 				err = module.PrepareTables()
-				if err != nil {
-					log.Errorw("failed to prepare tables", "error", err)
-					return err
-				}
+			}
+			if err != nil {
+				log.Errorw("failed to recreate tables", "error", err)
```
Comment (krish-nr) [inline modular/blocksyncer/blocksyncer_options.go:136]: log message 'recreate' should be replaced by migrate/prepare

File: modular/blocksyncer/blocksyncer_options.go [hunk 2]
```diff
@@ -249,50 +231,65 @@ func (b *BlockSyncerModular) getLatestBlockHeight(ctx context.Context) {
...
-				height := idx + count*c + startHeight
-				if height > uint64(latestBlockHeight) {
-					return
+		processedHeight, ok := <-Cast(b.parserCtx.Indexer).ProcessedQueue
+		if !ok {
+			log.Warnf("ProcessedQueue is closed")
+			return
+		}
+		log.Infof("processedHeight:%d, will process height:%d", processedHeight, count*cycle+startHeight)
+		if processedHeight != 0 && count*cycle+startHeight-processedHeight > 4*count {
```
Comment (krish-nr) [inline modular/blocksyncer/blocksyncer_options.go:249]: I think '4' here means block cache capacity for each worker, define this as a const value and name it like 'BlockCacheCapacity' or other meaningful names

**Review by 2129zxl — APPROVED** [review]
LGTM

**Review by krish-nr — APPROVED** [review]
LGTM

**Review by ruojunm — APPROVED** [review]

**Pull request merged**